service/src/main/java/bio/terra/workspace/service/workspace/exceptions/ChildrenBlockingDeletionException.java

```diff
@@ -0,0 +1,13 @@
+package bio.terra.workspace.service.workspace.exceptions;
+
+import bio.terra.common.exception.ErrorReportException;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+
+/** Exception thrown when a user attempts to delete a workspace that still has children */
+public class ChildrenBlockingDeletionException extends ErrorReportException {
+
+  public ChildrenBlockingDeletionException(String message, List<String> causes) {
+    super(message, causes, HttpStatus.CONFLICT);
```

Comment:
IETF status codes:

> [6.5.8](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.8).  409 Conflict
> 
>    The 409 (Conflict) status code indicates that the request could not
>    be completed due to a conflict with the current state of the target
>    resource.  This code is used in situations where the user might be
>    able to resolve the conflict and resubmit the request.  The server
>    SHOULD generate a payload that includes enough information for a user
>    to recognize the source of the conflict.
> 
>    Conflicts are most likely to occur in response to a PUT request.  For
>    example, if versioning were being used and the representation being
>    PUT included changes to a resource that conflict with those made by
>    an earlier (third-party) request, the origin server might use a 409
>    response to indicate that it can't complete the request.  In this
>    case, the response representation would likely contain information
>    useful for merging the differences based on the revision history.

I think that lines up, particularly when combined with the error message enumerating the resources that are blocking deletion.

Reply:
rawls returns 400.
I'm not sure consistency with rawls matters much here since:

1. It's still 4xx, just more semantic
2. this is after all the checks in rawls, so it's additive anyway
3. More specific error codes tend to make things easier to track down